index.js

```diff
@@ -55,9 +58,17 @@ module.exports = {
     require('./lib/routes')(self, options);
     require('./lib/api')(self, options);
   },
-  afterConstruct: function(self) {
+  afterConstruct: async function(self) {
     self.addRoutes();
     self.pushAsset('script', 'always', { when: 'lean' });
     self.pushAsset('stylesheet', 'dialog');
+
+    self.on('apostrophe-pages:beforeSend', 'addDialogs', async function() {
```

Comment:
This will fetch all dialog pieces on every page load in apostrophe .. feels very expensive. Plus, if a user has already seen a dialog box and their cookie is going to suppress the triggering anyway, then this is wasted effort.

We talked about this on Friday's call but also noting it here, we should only go and fetch relevant dialog boxes on the front-end dynamically, _after_ we've decided the user is eligible to see them (no cookie saying otherwise). This could be handled through calling a render route like sketched out here https://github.com/apostrophecms/dialog-sandbox/blob/master/lib/modules/sandbox/index.js#L5